### PR TITLE
DM-3649: Use mkstemp instead of tmpnam

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: tests/logTest.cc [hunk 1]
```diff
@@ -65,14 +67,35 @@
...
 
     ~LogFixture() {
-        unlink(ofName.c_str());
+        _unlink_tmp(ofName, fd);
+    }
+
+    std::string _tmpnam()
+    {
+        char cname[L_tmpnam];
+        strncpy(cname, P_tmpdir "logTest-XXXXXXXXX", sizeof(cname));
```
Comment (ktlim) [inline tests/logTest.cc]: I think you need a slash here; at least on my machine `P_tmpdir` is just `/tmp`.

Comment (timj) [inline tests/logTest.cc]: Thanks. I was worried about that. Didn't necessarily want to go full on `boost::filesystem`

File: tests/logTest.cc [hunk 2]
```diff
@@ -65,14 +67,35 @@
...
+        cname[sizeof(cname)-1] = '\0';  // Just in case of miscount of X
+        fd = mkstemp(cname);
```
Comment (ktlim) [inline tests/logTest.cc]: `mkstemp` is documented as requiring that the last six characters be "XXXXXX".  I don't think you guarantee this if `P_tmpdir` is too long.

Comment (timj) [inline tests/logTest.cc]: My memory management here is a bit fast and loose and relies on `L_tmpnam` being much larger than `P_tmpdir`. This is true on a Mac where `L_tmpnam` is 1024, and I assumed it was true on Linux, but now that I look on an NCSA linux machine I see that `L_tmpnam` may only be 20...

Reading the Linux man page for `mkstemp` is interesting as it behaves quite differently to the Mac version despite assertions of POSIX.

I'll revisit this a bit.

Comment (RobertLuptonTheGood) [inline tests/logTest.cc]: The posix specification http://pubs.opengroup.org/onlinepubs/9699919799/functions/mkdtemp.html doesn't make any guarantee about buffer sizes.  http://pubs.opengroup.org/onlinepubs/009695399/functions/tmpnam.html makes no statement about P_tmpdir (defined in http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdio.h.html). The man pages are not the standard!

**Pull request merged**